src/layouts/Layout.astro

```diff
@@ -0,0 +1,65 @@
+---
+interface Props {
+	title?: string;
+	description?: string;
+}
+
+import "../styles/globals.css";
+
+const { title, description } = Astro.props;
+
+const logoURL = new URL("/logo.png", Astro.url);
+
+const parsedTitle = title ? `${title} - Seren_Modz 21` : "Seren_Modz 21";
+const parsedDescription = description ?? "Welcome to Seren Modz's personal site! Here, you can find information about him, his projects, and more.";
+---
+
+<script is:inline>
+	const getThemePreference = () => {
+		if (typeof localStorage !== "undefined" && localStorage.getItem("theme")) {
+			return localStorage.getItem("theme");
+		}
+		return window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light";
+	};
+	const isDark = getThemePreference() === "dark";
+	document.documentElement.classList[isDark ? "add" : "remove"]("dark");
+ 
+	if (typeof localStorage !== "undefined") {
+		const observer = new MutationObserver(() => {
+			const isDark = document.documentElement.classList.contains("dark");
+			localStorage.setItem("theme", isDark ? "dark" : "light");
+		});
+		observer.observe(document.documentElement, { attributes: true, attributeFilter: ["class"] });
+	}
+</script>
+

```

Comment:
all of this file has tab intentation from create-astro template, which is not my thing